### ISSUE TITLE
Session Start step 6: reconciling the ledger from a pattern excerpt reports a drift the ledger itself already explains

### ISSUE THREAD

**Issue by thobi1987-prog** [rebelytics/one-skill-to-rule-them-all#168]
**Repo:** rebelytics/one-skill-to-rule-them-all
**Target:** `SKILL.md` → `## Session Start Protocol`, step 6 (*Targets and staged work*)
**Verified against upstream HEAD** (2026-09-14, `7518a85`): step 6 still reads
"If `skill-updates/PENDING.md` lists staged updates, reconcile the list before
announcing it", and says nothing about how the ledger is to be read.
**Adjacent, not duplicate:** #155 asks where the reconciliation takes its *scope*
from (the directory, not the ledger). This is about how the ledger is *read* once
it is in scope. #159 makes the neighbouring point for observations (compare bodies,
not titles); the same failure has a second site, in a procedure that is not about
observations at all.

## What happened

A session ran step 6 correctly in the sense #155 asks for: it enumerated the
staging directory for the scope and read `PENDING.md` only for the expectation.
It read the ledger with

```bash
grep -nE '^\s*(#{1,4}|\-|\*)\s' PENDING.md
```

— headings and list items, which is where staged rounds are recorded. It then
reported a drift to the user: the ledger lists round `X` for two skills, the
directory holds it for one.

The ledger explained this twenty lines further down, in prose the excerpt could not
contain: a previous session's keep-two prune had moved that copy to `_to_delete/`
and deleted it, and had written the classification and the outcome into the file.
The reported drift did not exist.

A second error rode along: the excerpt also omitted the line `Keine offenen
Einträge.` ("no open entries") — not a list item, so not in the output. The session
told the user the ledger needed clearing when it was already clear.

## Why an excerpt is structurally the wrong input here

Both misses have one shape. A pattern excerpt returns the entries that conform to
the pattern, and a hand-maintained ledger records its *exceptions* — resolutions,
prunes, one-off findings, a statement that nothing is pending — as prose that
deliberately does not look like an entry. So the excerpt is guaranteed to drop
exactly the content that would refute a drift, and never the content that suggests
one. The report is therefore biased toward false positives, and its confidence is
highest where its input is thinnest.

This is the same asymmetry #155 describes one level up (the silent direction is the
one nobody checks), reappearing inside a step that has already been fixed for
scope. Scope and reading depth are independent: taking the scope from the directory
does not make a partial read of the ledger safe.

## Suggested change

Add to step 6, next to the reconciliation:

> **Read the ledger whole before reporting a difference.** An excerpt — list
> lines, headings, the hits of a pattern — is a valid way to FIND an entry and
> never a valid input for a verdict about one: a hand-maintained file records its
> exceptions (a prune, a resolution, "no open entries") as prose that does not
> match the pattern its entries match, so the excerpt systematically drops the
> content that would refute a drift and keeps the content that suggests one.
> Excerpt to search, full text to judge.

## Generalisation

The rule is not specific to `PENDING.md`, and not to this skill: any comparison
between an artefact and a description of it is only as good as the more
discursive of the two was read. Where the description is maintained by hand, its
corrections live outside the format its entries follow — which is precisely why a
format-shaped read of it cannot see them.

**Comment by notbucki** [comment]
Second instance of the same shape, with the other symptom. Our ledger did not misreport a drift; it grew. Resolutions were written into it in place: entries the reconciliation had classified identical → installed (branch (a)) were re-headed "done" with the `diff` proof as a note and kept — eight of them by 2026-09-10 — and each later session read all of it at step 6. Branch (b) is the only one that asks for a note, and names no destination for it. Between 2026-09-02 and 2026-09-10 the file reached 824 lines / 67 KB with zero open entries, while every session read it whole, which is what the suggested sentence asks for. "Full text to judge" is the right rule, and its cost grows with exactly the prose it depends on; the gate's own text ("bind the cleanup to the moment the ledger is read") was in place the whole time and did not stop the growth.

What held, in case it helps the wording:

- **The ledger carries only open entries.** Notes, resolutions and reconciliation records move — byte-identical, nothing is lost — to a `HISTORY.md` beside it as soon as the reconciliation has classified the entry. Today the ledger is 1.2 KB against 63.5 KB of history. An excerpt and a whole read of the ledger then agree, because the file no longer records exceptions in a format its entries do not share.
- **A measurement at session start, not a rule in the file head.** A session-start hook warns when the ledger exceeds a size threshold or carries a resolved-entry heading, and the warning says "move the entries", not "acknowledge". The head rule ("remove the entry once installed") had been in the file throughout.

So two sentences rather than one: read the whole ledger to judge, and name where the note in (b) goes — a history file, not the ledger — so that "whole" stays a few kilobytes. Adjacent, not overlapping: #155 (scope from the directory) and #159 (per point) leave both untouched.

Checked against `main` (`7518a85`): SKILL.md step 6 and the reconciliation gate in `references/weekly-review.md` name the removal, not the note's destination.

Prepared with Claude Code (an AI coding agent) from one workspace ledger's history; a human reviewed and approved it. Codex (gpt-6-astra, effort max) reviewed the draft before posting: one broad round and three adversarial rounds; two P2 findings cut a branch attribution and a count the evidence did not carry.

**Comment by thobi1987-prog** [comment]
Agreed, and I can add a third instance from the same day this issue was filed — with the same fix arrived at independently, which is some evidence the fix is the natural one.

**What we found.** Our `PENDING.md` had reached 98 lines with *zero* open entries: every installed round had been kept as an HTML-commented block carrying its diff proof and keep-two notes. The false drift report in the issue body came from a pattern excerpt over exactly that file — the excerpt saw the block headings and not the prose that resolved them. So the two symptoms are one mechanism seen twice: resolutions written in place are prose the entry format does not share, and every reader then either skips it (excerpt → false drift) or pays for it (whole read → 824 lines).

**What we did, before seeing your comment.** Same split: the ledger keeps open entries only; everything the reconciliation classified as installed or superseded moved byte-identical to an `INSTALLED-LOG.md` beside it. Today: 2 KB ledger, 12 KB history. Since then a whole read and an excerpt agree, for the reason you give.

**On the wording.** Two sentences is right, and I'd tighten the second to name the trigger as well as the destination, so it binds to an event the session already performs:

> Read the ledger whole before reporting a difference — an excerpt is for finding an entry, never for a verdict about one. And keep it whole-readable: the ledger carries open entries only; the moment the reconciliation classifies an entry as installed or superseded, its record (the diff proof, the keep-two note, the resolution) moves byte-identical to a history file beside the ledger. A note written into the ledger in place is prose its entries do not share, and every later reader either skips it or pays for it.

That makes branch (b)'s note a *move*, not an *append*, and keeps "whole" at a few kilobytes without a size hook — though the hook is a good second layer, and I'd keep its phrasing exactly as you have it: "move the entries", not "acknowledge".

Adjacency as you say: #155 (scope from the directory) and #159 (per point) untouched.

Prepared with Claude Code from the workspace ledger and its history; the human owner reviewed and approved.

**Comment by rebelytics** [comment]
Thank you. Queued, and it'll go into one of the next weekly releases.
